technical_docs/Verification.md

```diff
@@ -0,0 +1,94 @@
+
+Each SciToken is issued by a single VO.  In order to verify the token itself, we must:
+1. Determine the VO the token claims to be issued by.
+2. Establish a set of trust roots to map a VO name to a set of known public keys.
+3. Demonstrate that the token was signed by a known public key.
+4. Determine the claims contained in the token follow the SciTokens domain-specific rules.
+
+Long-term, an important concept for SciTokens is the ability to delegate (creating a new token from an existing one with separate cryptographic material) and attenuate (limiting the scope of claims in a delegated) tokens.  This forms an ordered _token chain_ that can be validated as a whole.  To allow distributed verification and delegation together, it may be required that the bearer of a SciToken is able to prove posession of a particular private key.  Many ideas for SciToken verification have been borrowed from the X509 / PKI ecosystem: one important distinction is that tokens are generated (potentially with a private key) and sent to the remote party over a confidential, integrity-protected channel.  This is in contrast with an X509 certificate-signing request, where the recipient is the only entity in possession of the private key: this design decision is considered acceptable as the creator of the token always has a superset of the authorizations of the recipient.
+
+Support for token chaining may not be present in the initial SciTokens implementations; to understand the text in this document, a standalone SciToken may be considered a chain of length one.
+
+Locating a SciToken
+-------------------
+
+In addition to web services, An important use case for SciTokens is POSIX-like command line and batch system environments.  Hence, we believe it is important to have an unambiguous way for a POSIX process to locate a SciToken it should use for authorization with SciToken-aware services.
+
+Unless a token is explicitly specified by the end-user, SciToken client libraries should search the following locations for a SciToken file:
+* `SCITOKEN` environment variable.
+* On a POSIX-like platform, `/tmp/scitoken_u$ID`, where `$ID` is replaced by the POIX user ID.
+
+On POSIX-like platforms, the file must be owned by the effective UID.  The file is parsed in a line-oriented form; each non-empty line *not* starting with `#` is to be parsed as a base64-encoded SciTokens.  The client library should determine which token is to be used for each authorization.  (Author note: we should provide guidance on how to determine the most restrictive potentially-valid token available.)
+
+This document does not cover how a new SciToken might be generated or requested.
+
+Establishing Trust Roots
```

Comment:
I'd prefer to avoid out-of-band trust root management. I agree that a resource server must be configured with a set of trusted issuers (matching *iss* claims). Then to validate a token from a trusted issuer, let's use [oauth-discovery](https://tools.ietf.org/html/draft-ietf-oauth-discovery) to fetch the [jwks_uri](https://tools.ietf.org/html/rfc7517) containing the current JSON Web Key Set. All these things should be cached in some implementation-specific way, but I prefer to focus the interoperable specification on the [oauth-discovery](https://tools.ietf.org/html/draft-ietf-oauth-discovery) method of building up the cache.